publication-event-handlers/src/main/java/no/unit/nva/publication/events/bodies/DataEntryUpdateEvent.java

```diff
@@ -113,10 +110,21 @@ public boolean notEmpty() {
     @JsonProperty("topic")
     public String getTopic() {
         var type = extractDataEntryType();
-        return ENTRY_TYPE_TO_TOPIC_MAP.get(type);
+        return switch (type) {
+            case Resource resource -> RESOURCE_UPDATE_EVENT_TOPIC;
+            case DoiRequest doiRequest -> DOI_REQUEST_UPDATE_EVENT_TOPIC;
+            case PublishingRequestCase publishingRequestCase -> PUBLISHING_REQUEST_UPDATE_EVENT_TOPIC;
+            case Message message -> MESSAGE_UPDATE_EVENT_TOPIC;
+            case GeneralSupportRequest generalSupportRequest -> GENERAL_SUPPORT_REQUEST_UPDATE_EVENT_TOPIC;
+            case UnpublishRequest unpublishRequest -> UNPUBLISH_REQUEST_UPDATE_EVENT_TOPIC;
+            case FileEntry fileEntry -> this.isDeleteEvent()
+                                            ? FILE_ENTRY_DELETE_EVENT_TOPIC
+                                            : FILE_ENTRY_UPDATE_EVENT_TOPIC;
+            default -> throw new IllegalArgumentException("Unknown entry type: " + type);
+        };
```

Comment:
Refactoring this method and generating two types of events for file -> delete and update event, so there is no need for DeleteFileFromS3EventHandler to consume each update event and check if "newImage" is present.